File: 75 day challenge/day 03/code.py
from typing import List


class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        max_candies = max(candies)
        return [(candy + extraCandies >= max_candies) for candy in candies]

def main():
    solution = Solution()

    # Test Case 1
    candies1 = [2, 3, 5, 1, 3]
    extraCandies1 = 3
    result1 = solution.kidsWithCandies(candies1, extraCandies1)
    if result1 == [True, True, True, False, True]:
        print("Test Case 1 Passed")
    else:
        print(f"Test Case 1 Failed")

    # Test Case 2
    candies2 = [4, 2, 1, 1, 2]
    extraCandies2 = 1
    result2 = solution.kidsWithCandies(candies2, extraCandies2)
    if result2 == [True, False, False, False, False]:
        print("Test Case 2 Passed")
    else:
        print(f"Test Case 2 Failed")

    # Test Case 3
    candies3 = [12, 1, 12]
    extraCandies3 = 10
    result3 = solution.kidsWithCandies(candies3, extraCandies3)
    if result3 == [True, False, True]:
        print("Test Case 2 Passed")
    else:
        print(f"Test Case 3 Failed")


if __name__ == "__main__":
    main()

File: 75 day challenge/day 03/code.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
    vector<bool> kidsWithCandies(vector<int> &candies, int extraCandies)
    {
        int length = candies.size();
        vector<bool> ans(length, false);

        int max_candy = INT_MIN;
        for (int candy:candies)
        {
            max_candy = max(max_candy, candy);
        }
        for (size_t i = 0; i < length; i++)
        {
            ans[i] = (candies[i] + extraCandies >= max_candy);
        }

        return ans;
    }
};

int main()
{
    Solution solution;

    // Test Case 1
    vector<int> candies1 = {2, 3, 5, 1, 3};
    int extraCandies1 = 3;
    vector<bool> result1 = solution.kidsWithCandies(candies1, extraCandies1);
    cout << "Test Case 1 " << endl;
    cout << "Result: [true, true, true, false, true]" << endl;
    cout << "Actual Result: ";
    for (bool val : result1)
    {
        cout << boolalpha << val << " ";
    }
    cout << endl
         << endl;

    // Test Case 2
    vector<int> candies2 = {4, 2, 1, 1, 2};
    int extraCandies2 = 1;
    vector<bool> result2 = solution.kidsWithCandies(candies2, extraCandies2);
    cout << "Test Case 2 " << endl;
    cout << "Result: [true, false, false, false, false]" << endl;
    cout << "Actual Result: ";
    for (bool val : result2)
    {
        cout << boolalpha << val << " ";
    }
    cout << endl
         << endl;

    // Test Case 3
    vector<int> candies3 = {12, 1, 12};
    int extraCandies3 = 10;
    vector<bool> result3 = solution.kidsWithCandies(candies3, extraCandies3);
    cout << "Test Case 3 " << endl;
    cout << "Result: [true, false, true]" << endl;
    cout << "Actual Result: ";
    for (bool val : result3)
    {
        cout << boolalpha << val << " ";
    }
    cout << endl;

    return 0;
}
